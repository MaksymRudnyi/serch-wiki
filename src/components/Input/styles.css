.input-field {
    padding: 15px 20px;
    border: 1px solid grey;
    border-radius: 5px;
    font-size: 24px;
}