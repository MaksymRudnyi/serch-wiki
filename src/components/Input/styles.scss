.input-field {
    background: #fff;
    display: flex;
    border: 1px solid #dfe1e5;
    box-shadow: none;
    border-radius: 24px;
    z-index: 3;
    height: 44px;
    padding: 0 30px;
    font-size: 18px;
    width: 400px;

    &:focus {
        box-shadow: 0 1px 6px 0 rgba(32,33,36,0.28);
        border-color: rgba(223,225,229,0);
        outline: none;
    }
}