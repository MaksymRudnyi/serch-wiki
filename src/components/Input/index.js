import React from 'react';

import './styles.css'

const Input = ({placeholder, ...rest}) => <input className="input-field" placeholder={placeholder} {...rest}/>

export default Input;