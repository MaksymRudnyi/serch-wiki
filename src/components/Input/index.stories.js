import React from 'react';

import Input from '.';

export default {
  title: 'Components/Input',
  component: Input,
};

const Template = (args) => <Input {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  placeholder: 'Search data',
};
