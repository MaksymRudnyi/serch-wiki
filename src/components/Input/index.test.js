import React from 'react';
import { shallow } from 'enzyme'

import Input from '.';

const render = (props) => shallow(<Input {...props}/>);

describe('Input component', () => {
    let sut;
    let props;

    describe('without props', () => {
        beforeEach(() => {
            sut = render(props);
        });

        it('should match snapshot', () => {
            expect(sut).toMatchSnapshot();
        })
    });

    describe('with props', () => {
        beforeEach(() => {
            props = {
                placeholder: 'Input your data',
                name: 'search'
            };

            sut = render(props);
        });

        it('should match snapshot', () => {
            expect(sut).toMatchSnapshot();
        })
    });
});