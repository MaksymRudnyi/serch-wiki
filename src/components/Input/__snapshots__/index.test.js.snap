// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component with props should match snapshot 1`] = `
<input
  className="input-field"
  name="search"
  placeholder="Input your data"
/>
`;

exports[`Input component without props should match snapshot 1`] = `
<input
  className="input-field"
/>
`;
